functional-tests/fn_bridge_deposit_reclaim_drt_seen.py

```diff
@@ -0,0 +1,292 @@
+            self.logger.debug(f"Generating {blocks_to_generate} blocks in chunk {i+1}/{chunks}")
+            btcrpc.proxy.generatetoaddress(blocks_to_generate, UNSPENDABLE_ADDRESS)
+
+        # wait up a little bit
+        time.sleep(1)
+
+        # Make sure that the BTC refund address has the expected balance
+        refund_btc_balance = get_balance(
+            refund_addr,
+            btc_url,
+            btc_user,
+            btc_password,
+        )
+        self.logger.debug(f"Refund BTC balance (before takeback): {refund_btc_balance}")
+        assert refund_btc_balance == 0, "BTC balance is not zero"
+
+        # Check BTC balance of the recovery address
+        btc_recovery_balance_pre = get_balance_recovery(
+            recovery_addr,
+            bridge_pk,
+            btc_url,
+            btc_user,
+            btc_password,
+        )
+        self.logger.debug(f"DRT BTC balance (before takeback): {btc_recovery_balance_pre}")
+
+        # Spend the take back path
+        take_back_tx = bytes(
+            take_back_transaction(refund_addr, bridge_pk, btc_url, btc_user, btc_password)
+        ).hex()
+        self.logger.debug("Take back tx generated")
+
+        # Send the transaction to the Bitcoin network
+        txid = btcrpc.proxy.sendrawtransaction(take_back_tx)
+        self.logger.debug(f"sent take back tx with txid = {txid} for address {el_address}")
+        btcrpc.proxy.generatetoaddress(2, UNSPENDABLE_ADDRESS)
+        time.sleep(1)
+
+        # Make sure that the BTC recovery address has 0 BTC
+        btc_recovery_balance_post = get_balance_recovery(
+            recovery_addr,
+            bridge_pk,
+            btc_url,
+            btc_user,
+            btc_password,
+        )
+        self.logger.debug(f"DRT BTC balance (after takeback): {btc_recovery_balance_post}")
+        assert btc_recovery_balance_post == 0, "BTC balance is not zero"
+
+        # Make sure that the BTC refund address has the expected balance
+        refund_btc_balance_post = get_balance(
+            refund_addr,
+            btc_url,
+            btc_user,
+            btc_password,
+        )
+        self.logger.debug(f"Refund BTC balance (after takeback): {refund_btc_balance_post}")
+        expected_balance_lower_bound = btc_recovery_balance_pre - 100_000_000
+        self.logger.debug(f"Expected balance lower bound: {expected_balance_lower_bound}")
+        assert (
+            refund_btc_balance_post >= expected_balance_lower_bound
+        ), "BTC balance is not as expected"
+
+        # Now let's see if the EVM side has no funds
+        # Make sure that the el_address has zero balance
+        balance = int(rethrpc.eth_getBalance(f"0x{el_address}"), 16)
+        self.logger.debug(f"EVM balance (after takeback): {balance}")
+        self.logger.debug(f"Expected balance lower bound: {STRATA_DEPOSIT_LOWER_BOUND}")
+        assert balance >= STRATA_DEPOSIT_LOWER_BOUND, "EVM balance is not as expected"
```

Comment:
Shouldn't this assert that the balance in the EVM address is zero? A better approach is to check the difference in the EVM balance before and after the DRT is published (that difference must be zero).